executor/executor_test.go

```diff
@@ -2787,3 +2787,57 @@ func (s *testSuite) TestYearTypeDeleteIndex(c *C) {
 	tk.MustExec("delete from t;")
 	tk.MustExec("admin check table t")
 }
+
+func (s *testSuite) TestUnsignedDecimalOverflow(c *C) {
+	tests := []struct {
+		input  interface{}
+		hasErr bool
+		err    string
+	}{{
+		-1,
+		true,
+		"Out of range value for column",
+	}, {
+		"-1.1e-1",
+		true,
+		"Out of range value for column",
+	}, {
+		-1.1,
+		true,
+		"Out of range value for column",
+	}, {
+		-0,
+		false,
+		"",
+	},
+	}
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a decimal(10,2) unsigned)")
+	for _, t := range tests {
+		func() {
+			res, err := tk.Exec("insert into t values (?)", t.input)
+			if res != nil {
+				defer res.Close()
+			}
+			if t.hasErr {
+				c.Assert(err, NotNil)
+				c.Assert(strings.Contains(err.Error(), t.err), IsTrue)
+			} else {
+				c.Assert(err, IsNil)
+			}
+		}()
+	}
+
+	func() {
+		defer func() {
+			tk.MustExec("set sql_mode=default")
```

Comment:
We don't need to do it because the `tk` is created in this test case, it doesn't affect other test cases.